src/common/dtpthread.c

```diff
@@ -0,0 +1,52 @@
+#include <sys/resource.h>
+
+#include "common/dtpthread.h"
+
+gboolean dt_openmp_init_stacksize(void)
+{
+  pthread_attr_t attr;
+  size_t stacksize;
+  gboolean ret=TRUE;
+
+  pthread_attr_init(&attr);
+  pthread_attr_getstacksize(&attr, &stacksize);
+  
+  const gchar *openmp_stacksize_env=openmp_stacksize_env=g_getenv("OMP_STACKSIZE");
+  
+  gchar *s=g_strdup_printf("%dB", SAFESTACKSIZE);
+  
+  if (stacksize < SAFESTACKSIZE && g_strcmp0(openmp_stacksize_env, s) != 0 )
```

Comment:
OMP_STACKSIZE is a bit strange you can define 8*1024*1024 as "8192" or "8388608B" or "8M". I chose to define it as 8388608B and that is printed to stderr if the test failes